airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlCdcIntegrationTest.kt

```diff
@@ -4,253 +4,113 @@
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream
+import io.airbyte.protocol.models.v0.StreamDescriptor
+import io.airbyte.protocol.models.v0.SyncMode
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.sql.Connection
+import java.sql.Statement
+import org.junit.jupiter.api.BeforeAll
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Timeout
+import org.testcontainers.containers.MySQLContainer
 
-@MicronautTest(environments = [Environment.TEST], rebuildContext = true)
-@Disabled
 class MysqlCdcIntegrationTest {
-    @Inject
-    lateinit var supplier: ConfigurationJsonObjectSupplier<MysqlSourceConfigurationJsonObject>
 
-    @Inject lateinit var catalogFactory: ConfiguredCatalogFactory
+    @Test
+    fun testEmpty() {
+        val run1: BufferingOutputConsumer =
+            CliRunner.source("read", config(), configuredCatalog).run()
 
-    @Inject lateinit var inputStateFactory: InputStateFactory
+        connectionFactory.get().use { connection: Connection ->
+            connection.isReadOnly = false
+            connection.createStatement().use { stmt: Statement ->
+                stmt.execute("INSERT INTO test.tbl (k, v) VALUES (3, 'baz')")
+            }
+        }
 
-    @Test
-    @Property(name = "airbyte.connector.config.json", value = CDC_CONFIG_JSON)
-    @Property(name = "airbyte.connector.catalog.json", value = CDC_CATALOG)
-    fun testReadWithState() {
-        val pojoConfig: MysqlSourceConfigurationJsonObject = supplier.get()
-        val configuredCatalog = catalogFactory.make(CDC_CATALOG)
-        val list: List<io.airbyte.protocol.models.v0.AirbyteStateMessage>? =
-            ValidatedJsonUtils.parseList(
-                    io.airbyte.protocol.models.v0.AirbyteStateMessage::class.java,
-                    AIRBYTE_STATE_MESSAGE_JSON_2
-                )
-                // Discard states messages with unset type to allow {} as a valid input state.
-                .filter { it.type != null }
+        val run2InputState: List<AirbyteStateMessage> = listOf(run1.states().last())
+        val run2: BufferingOutputConsumer =
+            CliRunner.source("read", config(), configuredCatalog, run2InputState).run()
 
-        print(list)
-        // val inputState = inputStateFactory.make(AIRBYTE_STATE_MESSAGE)
-        // println(inputState)
-        // val output: BufferingOutputConsumer = CliRunner.runSource("read", pojoConfig,
-        // configuredCatalog)
-        val output: BufferingOutputConsumer =
-            CliRunner.source("read", pojoConfig, configuredCatalog, list).run()
-        Assertions.assertNotNull(output.specs())
+        println()
+        println()
+        for (msg in run1.messages()) {
+            println(Jsons.valueToTree(msg))
+        }
+        println()
+        for (msg in run2.messages()) {
+            println(Jsons.valueToTree(msg))
+        }
```

Comment:
I believe this new test to be roughly equivalent to the old one. If I missed out anything in particular please let me know.

Reply:
can you add a todo here to make sure we fill in test assertions?

Reply:
yes!! good call out

Reply:
done